### PR TITLE
dups_trimming.py: Updated the workflow to work with verification of Online Trimming

### PR DESCRIPTION
**Pull request by pdhiran**
1. Added a param to verify offline trimming.
- The existing test case always needed to start with build without the bug fix, Recreate the issue and then verify the Offline and Online trimming. With this change we are making the issue recreation optional.
2. updated the suite to mero nodes for testing.
- Updated the suite to use mero nodes as conf for testing
3. Updated the Stretch cluster suites both for downstream & upstream testing.
- Updated the deployment methods to use stretch mode deploy with spec file with loc attributes added during bootstrap.

### CODE REVIEW

**Comment by pdhiran** [comment]
Stretch Cluster pass logs : 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-Y7AL7P

Dups trimming :
Reef : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-NBYBJ9/

**Comment by neha-gangadhar** [comment]
Looks good, no major comments.

File: suites/pacific/rados/tier-2_rados_test-stretch-mode.yaml [hunk 1]
```diff
@@ -61,42 +121,28 @@ tests:
...
-      polarion-id: CEPH-83573758
+      polarion-id:
       config:
         command: add
-        id: client.1
-        node: node7
+        id: client.1                      # client Id (<type>.<Id>)
+        nodes:
+          - node8:
+              release: 6
```
Comment (harshkumarRH) [inline suites/pacific/rados/tier-2_rados_test-stretch-mode.yaml:135]: Is this a typo or do we intentionally want to install ceph client packages for RHCS 6 in Pacific test suite?

File: suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # Suite is to be used to verify the dups trimming scenario on all NVMe cluster
+# Conf file to be used : conf/quincy/baremetal_pipeline/mero_conf.yaml
```
Comment (harshkumarRH) [inline suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml:2]: Please change to `conf/reef/baremetal/mero_conf.yaml`

File: suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml
Comment (harshkumarRH) [inline suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml:1]: Please change client node from `depressa012` to `mero015`

File: suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml [hunk 3]
```diff
@@ -24,7 +25,7 @@ tests:
                 # using 6.1 rc candidate
                 custom_image: "registry-proxy.engineering.redhat.com/rh-osbs/rhceph:6-177"
                 custom_repo: "http://download-node-02.eng.bos.redhat.com/rhel-9/rel-eng/RHCEPH-6/RHCEPH-6.1-RHEL-9-20230602.0/compose/Tools/x86_64/os/"
```
Comment (harshkumarRH) [inline suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml:27]: As we are not following an upgrade path and PG Dups trimming verification is possible in a greenfield deployment, is it necessary to identify custom image of RHCS 6.1 rc build?

File: suites/quincy/rados/tier-2_rados_test-pg-dups-trimming.yaml [hunk 4]
```diff
@@ -24,7 +25,7 @@ tests:
                 # using 6.1 rc candidate
                 custom_image: "registry-proxy.engineering.redhat.com/rh-osbs/rhceph:6-177"
                 custom_repo: "http://download-node-02.eng.bos.redhat.com/rhel-9/rel-eng/RHCEPH-6/RHCEPH-6.1-RHEL-9-20230602.0/compose/Tools/x86_64/os/"
```
Comment (harshkumarRH) [inline suites/quincy/rados/tier-2_rados_test-pg-dups-trimming.yaml:27]: As we are not following an upgrade path and PG Dups trimming verification is possible in a greenfield deployment, is it necessary to identify custom image of RHCS 6.1 rc build?

File: suites/reef/rados/tier-2_rados_test-stretch-mode.yaml [hunk 5]
```diff
@@ -61,41 +121,28 @@ tests:
...
-      polarion-id: CEPH-83573758
+      polarion-id:
       config:
         command: add
-        id: client.1
-        node: node7
+        id: client.1                      # client Id (<type>.<Id>)
+        nodes:
+          - node8:
+              release: 6
```
Comment (harshkumarRH) [inline suites/reef/rados/tier-2_rados_test-stretch-mode.yaml:135]: If this is a mistake, please change release to 7

File: tests/rados/test_pg_dups_trimming.py [hunk 6]
```diff
@@ -38,6 +38,7 @@ def run(ceph_cluster, **kw) -> int:
     pool_configs = config["pool_configs"]
     pool_configs_path = config["pool_configs_path"]
     test_image = config.get("container_image")
+    verify_inflation = config.get("verify_inflation", True)
```
Comment (harshkumarRH) [inline tests/rados/test_pg_dups_trimming.py:41]: As verification of inflation of pg dups beyond the threshold is only possible with Pacific builds older than the fixed version, would it be better to keep the default value as `False`?

File: suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml (hunk 3 above)
Comment (harshkumarRH) [reply suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml:27]: Please disregard the above comment, it seems to be my misunderstanding, I did find the code to upgrade the cluster in test `tests/rados/test_pg_dups_trimming.py`

File: suites/quincy/rados/tier-2_rados_test-pg-dups-trimming.yaml (hunk 4 above)
Comment (harshkumarRH) [reply suites/quincy/rados/tier-2_rados_test-pg-dups-trimming.yaml:27]: Please disregard the above comment, it seems to be my misunderstanding, I did find the code to upgrade the cluster in test `tests/rados/test_pg_dups_trimming.py`

File: suites/pacific/rados/tier-2_rados_test-stretch-mode.yaml (hunk 1 above)
Comment (pdhiran) [reply suites/pacific/rados/tier-2_rados_test-stretch-mode.yaml:135]: Yes, was a miss from my side. Updated!

File: suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml (hunk 2 above)
Comment (pdhiran) [reply suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml:2]: Updated!

File: suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml
Comment (pdhiran) [reply suites/reef/rados/tier-2_rados_test-pg-dups-trimming.yaml:1]: Updated!

File: suites/reef/rados/tier-2_rados_test-stretch-mode.yaml (hunk 5 above)
Comment (pdhiran) [reply suites/reef/rados/tier-2_rados_test-stretch-mode.yaml:135]: Updated

File: tests/rados/test_pg_dups_trimming.py (hunk 6 above)
Comment (pdhiran) [reply tests/rados/test_pg_dups_trimming.py:41]: Sounds good. Updated

**Review by neha-gangadhar — APPROVED** [review]
Approving the PR.

**Pull request merged**